### DISCUSSION TITLE
Faulty STM32 chip or incorrect core code

### DISCUSSION THREAD

**Discussion by brightproject** [stm32duino/Arduino_Core_STM32#2076] [Q&A]
Good day @fpistm.
I use BlackPill `stm32f401ccu6`  and I need to work with `I2C` bus `PB6/PB7`.
Earlier I wrote an [issue](https://github.com/WeActStudio/WeActStudio.MiniSTM32F4x1/issues/44) about `BlackPill` hardware, because. possible damage to the `PA10` port.
But as soon as I connect the serial port `PA9/PA10` and the `I2C` bus, nothing works.
I'm starting to deal with serial ports.
The code is following
```
//                      RX    TX
HardwareSerial Serial3(PA10, PA9);
HardwareSerial Serial2(PB7, PB6);

void setup() {
  Serial3.begin(115200);
  Serial2.begin(115200);  
}

void loop() {
  Serial3.println("I am serial 3");
  Serial2.println("I am serial 2");
  delay(1000);
}
```
The code is just to check if serial ports are working.
This code was checked on 3 `BlackPill `and the work is equally wrong everywhere.
First, the code in the infinite loop is executed only 1 time.
![Скриншот 20-07-2023 12 49 15](https://github.com/stm32duino/Arduino_Core_STM32/assets/1788098/e38c07f5-0f07-49c6-9ce2-a80af97776a5)
In the screenshot below, each output is a separate press of the `RESET` button.
The output is single and without a line break, although the code indicates
`println`
When I connect `CH340` to the `PA9`(TX) leg, I get the output
`I am serial 3`
When I connect `CH340` to the `PB6`(TX) leg, I get exactly the same output
`I am serial 3`
And there is no line break.
What am I doing wrong?
PS.
If you specify `Serial1`, then the code does not compile at all ...
The compiler reports that the Serial1 variable has already been defined in the `HardwareSerial.cpp.o` file

> c:/users/admin/appdata/local/arduino15/packages/stmicroelectronics/tools/xpack-arm-none-eabi-gcc/10.3.1-2.3/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld.exe: C:\Users\admin\AppData\Local\Temp\arduino_cache_876671\core\core_71ce6b1b0fb14fa3999cc585d1e98a16.a(HardwareSerial.cpp.o):(.bss.Serial1+0x0): multiple definition of `Serial1'; sketch\Test.ino.cpp.o:(.bss.Serial1+0x0): first defined here
> collect2.exe: error: ld returned 1 exit status
> exit status 1

Code doesn't work if I set `define`
```
HardwareSerial Serial2(PA10, PA9);
  #define SerialPort Serial2
void setup() {
  SerialPort .begin(115200);  
}

void loop() {
  SerialPort .println("I am serial 2");
}
```
During compilation error also.

**Comment by fpistm** [comment]
Answered here #2077
